homeassistant/components/duotecno/climate.py

```diff
@@ -0,0 +1,83 @@
+"""Support for Duotecno climate devices."""
+from typing import Any
+
+from duotecno.protocol import SensFanspeed, SensPreset
+from duotecno.unit import SensUnit
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import DuotecnoEntity, api_call
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Velbus switch based on config_entry."""
+    cntrl = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        DuotecnoClimate(channel) for channel in cntrl.get_units(["SensUnit"])
+    )
+
+
+class DuotecnoClimate(DuotecnoEntity, ClimateEntity):
+    """Representation of a BinarySensor."""
+
+    _unit: SensUnit
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_modes = [HVACMode.OFF, HVACMode.HEAT, HVACMode.COOL]
+    _attr_preset_modes = list(SensPreset.__members__.keys())
+    _attr_fan_modes = list(SensFanspeed.__members__.keys())
```

Comment:
Isn't there a better way to get the modes instead of the `__members__.kets()`?

Reply:
If you have a suggestion I can change it, but I do not have a good idea

Reply:
Would just `list(SensPreset)` work as it's an enum?

Reply:
I don't think so

```
>>> list(SensFanspeed)
[<SensFanspeed.SPEED1: 0>, <SensFanspeed.SPEED2: 1>, <SensFanspeed.SPEED3: 2>, <SensFanspeed.SPEED4: 3>, <SensFanspeed.SPEED5: 4>, <SensFanspeed.AUTO: 255>]
>>> list(SensFanspeed.__members__.keys())
['SPEED1', 'SPEED2', 'SPEED3', 'SPEED4', 'SPEED5', 'AUTO']
```

Reply:
Do you want the names to be there hmmm

Reply:
What I see at other integrations is that they have a dict to transform between the integration specific modes and they use it for transforming it when getting the preset mode.

Reply:
we can do this, but this will require a bit more maintenance, when new presets are added we will need to add them to both locations, pyDuotecno and this integration.

With this system its only inside pyDuotecno, so upgrading pyDuotecno will get the new presets/fanspeeds automaticly in hass.

So i like this system better, but if its a dealbreaker i will change it

Reply:
Right, I asked some other people and we do like to have this mapping in HA. This way we can also add translations for the speeds (so I would suggest making the names use snake_case). So yes I would advice to do the mapping here